lib/libc/Kconfig

```diff
@@ -26,6 +34,10 @@ config PICOLIBC_SUPPORTED
 	help
 	  Selected when the target has support for picolibc.
 
+config NEWLIB_SUPPORTED
+	def_bool y
```

Comment:
Doesn't feel like this should be a `def_bool`.

Also, can we drop the `depends on` and just introduce something similar to cmake/toolchain/gnuarmemb/Kconfig

Reply:
> > Doesn't feel like this should be a `def_bool`.
> 
> That seems to be the pattern when you want to create a new symbol built by testing the state of others, but if you've an alternate preferred syntax, I'm happy to adjust.

Yeah, I think this should just be:
```
config NEWLIB_SUPPORTED
   bool
```

and leave it to other places to select it.

> 
> > Also, can we drop the `depends on` and just introduce something similar to cmake/toolchain/gnuarmemb/Kconfig
> 
> I tried that (placing a Kconfig in cmake/toolchain/zephyr) and it didn't work, perhaps because Zephyr SDK provides it's own Kconfig file? I didn't look into it further, once it failed.

We need to see if we can introduce a Kconfig file on the zephyr side that can co-exist w/the Kconfig from the SDK itself.

Reply:
Happy to adapt if there's an alternate solution available. Do you want to sort that out before the release, or get this bug fixed and then create a better solution later?

Reply:
> Yeah, I think this should just be:
> 
> ```
> config NEWLIB_SUPPORTED
>    bool
> ```
> 
> and leave it to other places to select it.

The current approach of making `NEWLIB_LIBC_SUPPORTED default y depends on "$(TOOLCHAIN_HAS_NEWLIB)" = "y"` is necessary for smoothing out the transition from the existing `TOOLCHAIN_HAS_NEWLIB` CMake variable/Kconfig to the new `NEWLIB_LIBC_SUPPORTED` Kconfig; otherwise, it will be a breaking change requiring Zephyr SDK and any out-of-tree toolchains to be aware of and properly configure the new `NEWLIB_LIBC_SUPPORTED` Kconfig.